export-mock-ddi-data.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+usage() {
+cat <<EOF
+Export a CSV file of Drug-Drug interactions from the Pathways system.
+
+Writes out to <outfile>
+
+Resultant CSV file is of the format:
+  drug_a, drug_b, {agonism or harmful}, time, unit
+
+Usage:
+    $0 <agonism|harmful> <outfile>
+EOF
+}
+
+
+loading() {
+    echo 'Exporting'
+    while true; do
+        printf "\r | "
+        sleep 0.125
+        printf "\r / "
+        sleep 0.125
+        printf "\r - "
+        sleep 0.125
+        printf "\r \\ "
+        sleep 0.125
+    done
+}
+
+
+main() {
+    if [ $# -ne 2 ]; then
+        usage
+        exit 1
+    fi
+
+    local readonly flavour="$1"
+    local readonly outfile="$2"
+
+    case "$flavour" in
+        harmful|agonism)
+            loading &
+            local readonly LOADING_SPINNER_PID="$!";
+            trap  "kill $LOADING_SPINNER_PID 2>&1 > /dev/null; exit" SIGTERM SIGINT
+            disown
+
+            sudo docker-compose \
+                -p export_ddis \
+                run asclepius python3 asclepius/enrich.py "$flavour" \
+                > "$outfile"
+
+            sudo docker-compose -p export_ddis down
+
+            kill "$LOADING_SPINNER_PID";
+            exit 0
+        ;;
+
+        *)
+            usage
+            exit 1
+        ;;
+    esac
+}
+
+if [ $EUID != 0 ]; then
+    sudo "$0" "$@"
```

Comment:
Ah, that's to get the pwd prompt asap, before launching the spinner. Otherwise there'll be some load of slashes and pipes being printed as user types.

I feel like this spinner is adding much complexity. 🤔